graphite/graphite_tcp.go

```diff
@@ -95,11 +102,35 @@ func (t *TCPServer) handleConnection(conn net.Conn) {
 	defer conn.Close()
 	defer t.wg.Done()
 
+	batcher := influxdb.NewPointBatcher(t.batchSize, t.batchTimeout)
+	in, out := batcher.Start()
 	reader := bufio.NewReader(conn)
+
+	// Start processing batches.
+	var wg sync.WaitGroup
+	done := make(chan struct{})
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		for {
+			select {
+			case batch := <-out:
+				_, e := t.writer.WriteSeries(t.database, "", batch)
+				if e != nil {
+					t.Logger.Printf("failed to write point batch to database %q: %s\n", t.database, e)
+				}
+			case <-done:
+				return
+			}
+		}
+	}()
+
 	for {
 		// Read up to the next newline.
 		buf, err := reader.ReadBytes('\n')
 		if err != nil {
+			close(done)
```

Comment:
Do we need some type of `flusher` on the batcher so that when we shut down we can ensure it drains the batch before we exit?